agent/api/task_unix.go

```diff
@@ -25,16 +33,113 @@ const (
 	//memorySwappinessDefault is the expected default value for this platform. This is used in task_windows.go
 	//and is maintained here for unix default. Also used for testing
 	memorySwappinessDefault = 0
+
+	defaultCPUPeriod = 100 * time.Millisecond // 100ms
+	// With a 100ms CPU period, we can express 0.01 vCPU to 10 vCPUs
+	maxTaskVCPULimit = 10
+	// Reference: http://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ContainerDefinition.html
+	minimumCPUShare = 2
 )
 
-func (task *Task) adjustForPlatform() {}
+func (task *Task) adjustForPlatform(cfg *config.Config) {
+	task.memoryCPULimitsEnabledLock.Lock()
+	defer task.memoryCPULimitsEnabledLock.Unlock()
+	task.memoryCPULimitsEnabled = cfg.TaskCPUMemLimit
+}
 
 func getCanonicalPath(path string) string { return path }
 
-// CgroupEnabled checks whether cgroups need be enabled at the task level
-func (task *Task) CgroupEnabled() bool {
-	//stub for now
-	return false
+// BuildCgroupRoot helps build the task cgroup prefix
+// Example: /ecs/task-id
+func (task *Task) BuildCgroupRoot() (string, error) {
+	taskID, err := task.GetID()
+	if err != nil {
+		return "", errors.Wrapf(err, "task build cgroup root: unable to get task-id from task ARN: %s", task.Arn)
+	}
+
+	return filepath.Join(config.DefaultTaskCgroupPrefix, taskID), nil
 }
 
-func (task *Task) platformHostConfigOverride(hostConfig *docker.HostConfig) {}
+// BuildLinuxResourceSpec returns a linuxResources object for the task cgroup
+func (task *Task) BuildLinuxResourceSpec() (specs.LinuxResources, error) {
+	linuxResourceSpec := specs.LinuxResources{}
+
+	// If task level CPU limits are requested, set CPU quota + CPU period
+	if !utils.ZeroOrNil(task.VCPULimit) {
```

Comment:
(multiple occurrences) I would prefer to avoid `ZeroOrNil` in critical code.  I think it would be better if you explicitly checked for those things here.  `ZeroOrNil` is really meant for handling unknown data (`interface{}`) and not really for handling things where the type is known.